tests/bot/cogs/test_snekbox.py

```diff
@@ -0,0 +1,363 @@
+        )
+        self.cog.get_status_emoji.assert_has_calls([call({'stdout': '', 'returncode': 0})] * 2)
+        self.cog.get_results_message.assert_has_calls([call({'stdout': '', 'returncode': 0})] * 2)
+        self.cog.format_output.assert_has_calls([call('')] * 2)
+
+        self.bot.wait_for.has_calls(
+            call('message_edit', check=snekbox.predicate_eval_message_edit, timeout=10),
+            call('reaction_add', check=snekbox.predicate_eval_emoji_reaction, timeout=10)
+        )
+        ctx.message.add_reaction.assert_called_once_with('🔁')
+        ctx.message.clear_reactions.assert_called()
+        response_msg.delete.assert_called_once()
+
+    def test_predicate_eval_message_edit(self):
+        """Test the predicate_eval_message_edit function."""
+        msg0 = MockMessage()
+        msg0.id = 1
+        msg0.content = 'abc'
+        msg1 = MockMessage()
+        msg1.id = 2
+        msg1.content = 'abcdef'
+        msg2 = MockMessage()
+        msg2.id = 1
+        msg2.content = 'abcdef'
+
+        cases = (
+            (msg0, msg0, False, 'same ID, same content'),
+            (msg0, msg1, False, 'different ID, different content'),
+            (msg0, msg2, True, 'same ID, different content')
+        )
+        for ctx_msg, new_msg, expected, testname in cases:
+            with self.subTest(msg=f'Messages with {testname} return {expected}'):
+                ctx = MockContext()
+                ctx.message = ctx_msg
+                self.assertEqual(snekbox.predicate_eval_message_edit(ctx, ctx_msg, new_msg), expected)
+
+    def test_predicate_eval_emoji_reaction(self):
+        """Test the predicate_eval_emoji_reaction function."""
+        valid_reaction = MockReaction()
+        valid_reaction.message.id = 1
+        valid_reaction.__str__.return_value = '🔁'
+        valid_ctx = MockContext()
+        valid_ctx.message.id = 1
+        valid_ctx.author.id = 2
+        valid_user = MockUser()
+        valid_user.id = 2
+
+        invalid_reaction_id = MockReaction()
+        invalid_reaction_id.message.id = 42
+        invalid_reaction_id.__str__.return_value = '🔁'
+        invalid_user_id = MockUser()
+        invalid_user_id.id = 42
+        invalid_reaction_str = MockReaction()
+        invalid_reaction_str.message.id = 1
+        invalid_reaction_str.__str__.return_value = ':longbeard:'
+
+        cases = (
+            (invalid_reaction_id, valid_user, False, 'invalid reaction ID'),
+            (valid_reaction, invalid_user_id, False, 'invalid user ID'),
+            (invalid_reaction_str, valid_user, False, 'invalid reaction __str__'),
+            (valid_reaction, valid_user, True, 'matching attributes')
+        )
+        for reaction, user, expected, testname in cases:
+            with self.subTest(msg=f'Test with {testname} and expected return {expected}'):
+                self.assertEqual(snekbox.predicate_eval_emoji_reaction(valid_ctx, reaction, user), expected)
```

Comment:
Reads better as separate lines.

```suggestion
                actual = snekbox.predicate_eval_emoji_reaction(valid_ctx, reaction, user)
                self.assertEqual(actual, expected)
```